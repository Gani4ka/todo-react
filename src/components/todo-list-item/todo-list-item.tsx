/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';

import './todo-list-item.css';

interface TodoListItemProps {
  label: string;
  onDeleted: React.MouseEventHandler;
  onToggleImportant: React.MouseEventHandler;
  onToggleDone: React.MouseEventHandler;
  important: boolean;
  done: boolean;
}

export const TodoListItem = (props: TodoListItemProps): JSX.Element => {
  const {
    label,
    onDeleted,
    onToggleImportant,
    onToggleDone,
    important,
    done
  } = props;

  let classNames = 'todo-list-item';
  if (done) {
    classNames += ' done';
  }

  if (important) {
    classNames += ' important';
  }

  return (
    <span className={classNames}>
      <span className='todo-list-item-label' onClick={onToggleDone}>
        {label}
      </span>

      <button
        type='button'
        className='btn btn-outline-success btn-sm float-right'
        onClick={onToggleImportant}
      >
        <i className='fa fa-exclamation' />
      </button>

      <button
        type='button'
        className='btn btn-outline-danger btn-sm float-right'
        onClick={onDeleted}
      >
        <i className='fa fa-trash-o' />
      </button>
    </span>
  );
};
