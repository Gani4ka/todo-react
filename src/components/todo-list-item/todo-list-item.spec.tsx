import React from 'react';
import { shallow } from 'enzyme';
import { TodoListItem, TodoListItemProps } from './todo-list-item';

describe('<TodoListItem />', (): void => {
  it('should render correctly', (): void => {
    const props: TodoListItemProps = {
      label: 'text',
      onDeleted: jest.fn(),
      onToggleImportant: jest.fn(),
      onToggleDone: jest.fn(),
      important: false,
      done: false
    };
    const component = shallow(<TodoListItem {...props} />);
    expect(component.debug()).toMatchSnapshot();
    // expect(component).toMatchSnapshot();
  });
});
