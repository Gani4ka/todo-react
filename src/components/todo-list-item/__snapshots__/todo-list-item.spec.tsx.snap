// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem /> should render correctly 1`] = `
"<span className=\\"todo-list-item\\">
  <span className=\\"todo-list-item-label\\" onClick={[Function: mockConstructor]}>
    text
  </span>
  <button type=\\"button\\" className=\\"btn btn-outline-success btn-sm float-right\\" onClick={[Function: mockConstructor]}>
    <i className=\\"fa fa-exclamation\\" />
  </button>
  <button type=\\"button\\" className=\\"btn btn-outline-danger btn-sm float-right\\" onClick={[Function: mockConstructor]}>
    <i className=\\"fa fa-trash-o\\" />
  </button>
</span>"
`;
